Sync/Synchronizers/ClientsSynchronizer.swift

```diff
@@ -158,6 +177,28 @@ public class ClientsSynchronizer: BaseSingleCollectionSynchronizer, Synchronizer
         return deferResult((false, []))
     }
 
+    private func uploadRecords(should: Bool, ifUnmodifiedSince: Timestamp?, toLocalClients localClients: RemoteClientsAndTabs, withServer storageClient: Sync15CollectionClient<ClientPayload>) -> Success {
+        return localClients.getCommands() >>== { clients in
+            for client in clients {
+                storageClient.get(client.guid!) >>== { response in
+                    let record = response.value
+                    if var clientRecord = record.payload.asDictionary {
+                        clientRecord["commands"] = JSON(record.payload.commands + (client.commands?.map { JSON($0.value) } ?? []))
+                        storageClient.put(Record(id: client.guid!, payload: ClientPayload(JSON(clientRecord)), ttl: ThreeWeeksInSeconds), ifUnmodifiedSince: record.modified)
+                            >>== { resp in
+                                log.debug("Client record upload succeeded.")
+                                localClients.deleteCommands(client.guid!)
+                        }
+                    }
+                }
+            }
+            return succeed()
```

Comment:
This is a little bit of a code smell. We should fail if an upload fails. So you probably want either:

```
return all(clients.map { storageClient.get( … 
```

or to use `walk`.